File: java/arcade/BishopAndPawn.java
public final class BishopAndPawn {

	boolean bishopAndPawn(String bishop, String pawn) {
		int x1 = bishop.charAt(0) - 'a',
	        y1 = Integer.valueOf(bishop.charAt(1)+"").intValue() - 1,
	        x2 = pawn.charAt(0) - 'a',
	        y2 = Integer.valueOf(pawn.charAt(1)+"").intValue() - 1;
	    for(int i = 0, x = x1, y = y1; i < 4; i++, x = x1, y = y1) {
	        for(int j = 0; j < 10; j++) {
	            if(x == x2 && y == y2) { return true; }
	            switch(i) {
	                case 0:
	                    x++;
	                    y++;
	                    break;
	                case 1:
	                    x++;
	                    y--;
	                    break;
	                case 2:
	                    x--;
	                    y--;
	                    break;
	                case 3:
	                    x--;
	                    y++;
	            }
	        }
	    }
	    return false;
	}
}

File: Arcade/Intro/Ruby/bishopAndPawn.rb
def bishopAndPawn(bishop, pawn)
	x1, y1 = bishop[0].ord - 97, bishop[1].to_i - 1
    x2, y2 = pawn[0].ord - 97, pawn[1].to_i - 1
    4.times do |i|
    	 x, y = x1, y1
    	 10.times do |j|
	    	return true if x == x2 and y == y2
	    	case i
	    		when 0
	    			x, y = x + 1, y + 1
	    		when 1
	    			x, y = x + 1, y - 1
	    		when 2
	    			x, y = x - 1, y - 1
	    		when 3
	    			x, y = x - 1, y + 1
	    	end
	    end
    end
    return false
end